src/main/java/org/apache/maven/shared/filtering/DefaultMavenFileFilter.java

```diff
@@ -110,7 +110,8 @@ public void copyFile( File from, File to, boolean filtering, List<FileUtils.Filt
         }
         catch ( IOException e )
         {
-            throw new MavenFilteringException( e.getMessage(), e );
+            throw new MavenFilteringException( ( filtering ? "filtering " : "copying " ) + from.getPath() + " to "
```

Comment:
What is the purpose to duplicate information like here?

Reply:
What do you mean with duplicate information? When maven-resources-plugin currently fails, it only displays a cryptic error message like `[ERROR] Failed to execute goal org.apache.maven.plugins:maven-resources-plugin:3.2.0:resources (default-resources) on project test: Input length = 1 -> [Help 1]` without **any** context information. Even when using `mvn -e` the displayed stacktrace does not show any information about the file that failed during the operation.

Reply:
The the root cause does not bubble up, then this is crap. Likely the log message is not ok. You should not try for this the symptom here. Please look in MRESOURCES where this exception is handled.

Reply:
The maven resources plugin is delegating the iteration over files to MavenResourcesFiltering, thus is not aware of the file for which filtering fails and therefore cannot report the file path.

Adding the file paths to the error message here has the advantage that all potential code paths to DefaultMavenFileFilter will be covered.

Reply:
To illustrate the effect of this PR some screenshots:

1) error message before this PR

![image](https://user-images.githubusercontent.com/426959/104460261-23e7db00-55ae-11eb-8597-4ea95c5098dc.png)

2) error message with this PR

![image](https://user-images.githubusercontent.com/426959/104460295-33672400-55ae-11eb-801a-091a62190773.png)